applications/asset_tracker_v2/src/modules/location_module.c

```diff
@@ -0,0 +1,634 @@
+
+	if (methods_index_cellular != -1) {
+		config.methods[methods_index_cellular].cellular.timeout =
+			DATA_FETCH_TIMEOUT_NEIGHBORHOOD_SEARCH * MSEC_PER_SEC;
+	}
+
+	LOG_DBG("Requesting location...");
+
+	err = location_request(&config);
+	if (err) {
+		SEND_EVENT(location, LOCATION_MODULE_EVT_DATA_NOT_READY);
+		LOG_ERR("Location request failed: %d", err);
+		return;
+	}
+	SEND_EVENT(location, LOCATION_MODULE_EVT_ACTIVE);
+	stats.start_uptime = k_uptime_get();
+}
+
+static void inactive_send(void)
+{
+	SEND_EVENT(location, LOCATION_MODULE_EVT_INACTIVE);
+}
+
+static void time_set(void)
+{
+	/* Change datetime.year and datetime.month to accommodate the
+	 * correct input format.
+	 */
+	struct tm gnss_time = {
+		.tm_year = pvt_data.datetime.year - 1900,
+		.tm_mon = pvt_data.datetime.month - 1,
+		.tm_mday = pvt_data.datetime.day,
+		.tm_hour = pvt_data.datetime.hour,
+		.tm_min = pvt_data.datetime.minute,
+		.tm_sec = pvt_data.datetime.seconds,
+	};
+
+	date_time_set(&gnss_time);
+}
+
+static inline int adjust_rsrp(int input)
+{
+	if (IS_ENABLED(CONFIG_LOCATION_MODULE_NEIGHBOR_CELLS_DATA_CONVERT_RSRP_TO_DBM)) {
+		return RSRP_IDX_TO_DBM(input);
+	}
+
+	return input;
+}
+
+static inline int adjust_rsrq(int input)
+{
+	if (IS_ENABLED(CONFIG_LOCATION_MODULE_NEIGHBOR_CELLS_DATA_CONVERT_RSRQ_TO_DB)) {
+		return round(RSRQ_IDX_TO_DB(input));
+	}
+
+	return input;
+}
+
+static void send_neighbor_cell_update(struct lte_lc_cells_info *cell_info)
+{
+	struct location_module_event *evt = new_location_module_event();
+
+	BUILD_ASSERT(sizeof(evt->data.neighbor_cells.cell_data) ==
+		     sizeof(struct lte_lc_cells_info));
+	BUILD_ASSERT(sizeof(evt->data.neighbor_cells.neighbor_cells) >=
+		     sizeof(struct lte_lc_ncell) * CONFIG_LTE_NEIGHBOR_CELLS_MAX);
+
+	memcpy(&evt->data.neighbor_cells.cell_data, cell_info, sizeof(struct lte_lc_cells_info));
+	memcpy(&evt->data.neighbor_cells.neighbor_cells, cell_info->neighbor_cells,
+	       sizeof(struct lte_lc_ncell) * evt->data.neighbor_cells.cell_data.ncells_count);
+
+	/* Convert RSRP to dBm and RSRQ to dB per "nRF91 AT Commands" v1.7. */
+	evt->data.neighbor_cells.cell_data.current_cell.rsrp =
+			adjust_rsrp(evt->data.neighbor_cells.cell_data.current_cell.rsrp);
+	evt->data.neighbor_cells.cell_data.current_cell.rsrq =
+			adjust_rsrq(evt->data.neighbor_cells.cell_data.current_cell.rsrq);
+
+	for (size_t i = 0; i < evt->data.neighbor_cells.cell_data.ncells_count; i++) {
+		evt->data.neighbor_cells.neighbor_cells[i].rsrp =
+			adjust_rsrp(evt->data.neighbor_cells.neighbor_cells[i].rsrp);
+		evt->data.neighbor_cells.neighbor_cells[i].rsrq =
+			adjust_rsrq(evt->data.neighbor_cells.neighbor_cells[i].rsrq);
+	}
+
+	evt->type = LOCATION_MODULE_EVT_NEIGHBOR_CELLS_DATA_READY;
+	evt->data.neighbor_cells.timestamp = k_uptime_get();
+
+	APP_EVENT_SUBMIT(evt);
+}
+
+/* Non-static so that this can be used in tests to mock location library API. */
+void location_event_handler(const struct location_event_data *event_data)
```

Comment:
```suggestion
static void location_event_handler(const struct location_event_data *event_data)
```

Instead of declaring it non-static, you can do something like this in your unit test

In unit test file: 
```
static int location_init_callback(location_event_handler_t event_handler, int no_of_calls)
{
	/* Copy in the event handler for future use. */
	location_event_handler = event_handler;
	return 0;
}

int test_foo(){
   ...
   __wrap_location_init_AddCallback(&location_init_callback);

   call_the_function_under_test();

   location_event_handler(whatever);
   ...
}

```

Reply:
Yep, it's possible but which one is more preferred. Leaving one function non-static and then letting all possible tests to use it directly with a declaration `extern void location_event_handler(const struct location_event_data *event_data);`, or having additional function (`location_init_callback()`) and its call (`__wrap_location_init_AddCallback(&location_init_callback);`) in all tests mocking this API.

I've done this non-static version somewhere else too, like in at_monitor (`at_monitor_dispatch()`), which should be particularly common for mocking.

Reply:
Discussed offline and agree with how it is now.